drizzlepac/haputils/product.py

```diff
@@ -657,7 +658,41 @@ def wcs_drizzle_product(self, meta_wcs):
             )
         )
 
-        edp_filenames = [element.full_filename for element in self.edp_list]
+        # Determine if there are any single exposure filter images which 
+        # should NOT be used in the computation of the total detection image in
+        # order to minimize cosmic ray contamination
+        #
+        # Determine the unique filters in the total product and create a
+        # dictionary to hold the filter name and the number of exposures per filter
+        set_of_filters = set([element.filters for element in self.edp_list])
+        count_dict = {key: [] for key in set_of_filters}
+
+        # Loop over the exposure objects to collect the exposures for each filter
+        for expo in self.edp_list:
+            for key in count_dict:
+                if key == expo.filters:
+                    count_dict[key].append(expo.full_filename)
+                    break
+
+        # Accumulate the exposure file names for only the filters which have
+        # more than one exposure
+        exclude_string = []
+        edp_filenames = []
+        for key, value in count_dict.items():
+            if len(value) > 1:
+                edp_filenames += value
+            else:
+                exclude_string.append("Excluding single exposure filter image {} for {}.".format(value, key))
+
+        # However, if all the filters only have one exposure, just use all
+        # the exposures
+        if not edp_filenames:
+            for value in count_dict.values():
+                edp_filenames += value
+        else:
```

Comment:
I think it will be useful to have a log entry here for this unusual case.

```suggestion
            log.info("All filters are single exposures, so all are included.")
        else:
```

Reply:
Good idea.